src/main/java/org/subethamail/smtp/server/SessionHandler.java

```diff
@@ -0,0 +1,11 @@
+package org.subethamail.smtp.server;
+
+import org.subethamail.smtp.DropConnectionException;
+
+public interface SessionHandler {
+
+    public void acquire(Session session) throws DropConnectionException;
```

Comment:
This is not very clear to me.
What do you mean with 'acquire' ?
I see that the impl in SourceTrackingSessionHandler is about concurrency but this sounds like a more general interface.
What about "onSessionStart" ?

If we don't go with a generic interface SessionHandler I suggest to use a narrower name like 'SessionTracker' or ConcurrencyHandler or something like that